hibernate-reactive-core/src/main/java/org/hibernate/reactive/pool/impl/ReactiveConnectionPoolInitiator.java

```diff
@@ -40,7 +40,7 @@ public ReactiveConnectionPool initiateService(Map configurationValues, ServiceRe
 		Object configValue = configurationValues.get( Settings.SQL_CLIENT_POOL );
 
 		String dbType = (String) configurationValues.get( "db" );
-		if ( dbType.equalsIgnoreCase( "h2" ) ) {
+		if ( dbType != null && dbType.equalsIgnoreCase( "h2" ) ) {
```

Comment:
The example tests apparently don't end up setting the type?

Reply:
Wait... this is completely wrong. dbType is something that we use for testing. It's not supposed to be used in the pool initiator

Reply:
This class defines the client connection pool.  I couldn't find any other way to search the service registry and select the H2SqlClientPool.

Reply:
I think we need to register a new service when we add H2 on the class path

Reply:
I actually suspect we could create a generic JDBC module that people could use if they want to test with any jdbc driver, but let's keep it as a follow up issue

Reply:
I think the right approach would be to override the service when the H2 dependency is on the class path.
But I don't know exactly how to do that on top of my head. We probably need to implement the `ServiceContributor` interface.

Reply:
> I actually suspect we could create a generic JDBC module that people could use if they want to test with any jdbc driver

I'm not very keen on that.

Giving them _a_ way to test in-memory (on H2) is reasonable, but what is the real use case for being able to pretend an arbitrary JDBC driver is reactive? I don't see it.

It doesn't even really work in terms of testing, since if the underlying driver is blocking, you're testing something pretty different from what's going to happen in prod.

Reply:
@DavideD 
`ReactiveServiceInitiators` holds a collection of initiators, of  which `ReactiveConnectionPoolInitiator.INSTANCE` is added.

This was the option I could think of to inject/override `DefaultSqlClientPool()`

Thoughts, or ideas?

Reply:
It was just an idea, but I don't see much difference in supporting the  H2 JDBC driver or a more generic driver. Most of the changes we are going to make are not specific to H2 (we just set the dialect and some transitive dependencies as default I think). I suspect we would just need to rename the classes and people could use any driver they want.

This would really be a bridge between Hibernate Reactive and the [Vert.x JDBC client](https://vertx.io/docs/vertx-jdbc-client/java/). The fact that one needs to use a different dependency and it might require extra configuration should be enough to let them know that this is different than Hibernate Reactive Core.
To me, this plus some documentation is enough to not feel responsible if people don't understand what they are doing.

That said, I'm not sure if it would work when having several jdbc drivers on the classpath or if there is a jbdc driver and a vertx driver. Maybe  it's not worth the hassle.

> It doesn't even really work in terms of testing, since if the underlying driver is blocking, you're testing something pretty different from what's going to happen in prod.

This is also true for H2, but we are assuming that people will understand this when they are going to use our integration.

Reply:
> Most of the changes we are going to make are not specific to H2

OK perhaps I'm wrong.

Reply:
There are a couple more databases which are primarily used "in memory" and geared more towards testing purposes than production use; it might be useful to maintain this code in a slightly generic fashion to *allow* such usage without necessarily recommending them.

We can certainly be more restrictive with the options we expose in Quarkus; e.g. I totally expect it to be "H2 exclusively" - at least initially - and we'll be hard wiring things both to make it easier to setup but also to avoid unintended use.